### ISSUE TITLE
Failing test

### ISSUE THREAD

**Issue by TheTincho** [prometheus/prometheus#1962]
Hi,

When building prometheus 1.1.1 for Debian, I have one test failing:

```
=== RUN   TestTargetSetRecreatesTargetGroupsEveryRun
time="2016-09-08T09:01:12Z" level=error msg="Error refreshing DNS targets: could
 not resolve srv.name.one.example.org: No server responded" source="dns.go:118" 
--- FAIL: TestTargetSetRecreatesTargetGroupsEveryRun (5.00s)
        targetmanager_test.go:36: 'dns/0/srv.name.one.example.org' should be present in TargetSet.tgroups: map[]

FAIL
exit status 1
FAIL    github.com/prometheus/prometheus/retrieval      8.587s
```

I don't know if this is really trying to access the network or not. In any case, it is a problem: Debian builds are forbidden from trying to access the network at all.

This is running with golang 1.7

**Comment by beorn7** [comment]
Perhaps the test wants the negative response from the network that srv.name.one.example.org doesn't exist.

And yes, that's something we should avoid.

**Comment by grobie** [comment]
@dmilstein it seems you added that test. Do you have time to rewrite it to not depend on a internet connection / working DNS discovery?

**Comment by dmilstein** [comment]
Drat, didn't realize it was actually resolving. Yep, I'll take a look and fix that. 

Sorry about that,
D

iPhone brevity in effect

**Comment by grobie** [comment]
Fixed with #1966
